java/FoodTruckApp2/src/main/java/com/my/FoodTruckApp2/Customer/CustomerService.java

```diff
@@ -29,4 +34,17 @@ public void createNewCustomer(){
             System.out.println("A new row has been inserted!!!");
         }
     }
+// ---------GETTING CUSTOMER BY THEIR ID----------//
+
+    public Customer gettingCustomerById(@PathVariable Integer id){
+        String sql = "SELECT * FROM customer WHERE id = ?";
+        try{
+         //this is an example of Mapping: we are getting the string but it then reads essentially the string and places it the right fields(id etc.) for the object we have in java.
+            Customer customerId = jdbcTemplate.queryForObject(sql,new BeanPropertyRowMapper<>(Customer.class),id);
+            // queryForObject return the whole object(A specific object)
+            return customerId;
+        }catch (EmptyResultDataAccessException emptyResultDataAccessException){
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND);
+        }
+    }
```

Comment:
can we delete these comments, please?

Reply:
can we add a log.error in the catch block?

additionally, can we add a message within the `ResponseStatusException`? (Second parameter)

Reply:
The comments won't be deleted but i updated the ResponseStatusException and added the log.error